src/ert/experiment_server/_server.py

```diff
@@ -60,29 +65,31 @@ async def handle_dispatch(
         self, websocket: WebSocketServerProtocol, path: str
     ) -> None:
         """handle_dispatch(self, websocket, path: str)
-
         Handle incoming "dispatch" connections, which refers to remote workers.
-
         websocket is a https://websockets.readthedocs.io/en/stable/reference/server.html#websockets.server.WebSocketServerProtocol  # pylint: disable=line-too-long
         """
+        event: Union[CloudEvent, DispatcherMessage]
         async for msg in websocket:
             if isinstance(msg, bytes):
-                # TODO handle protobuf messages; update statemachine
-                event_logger.debug("handle_dispatch pbuf: %s", msg)
-                continue
-            try:
-                event = from_json(msg, data_unmarshaller=evaluator_unmarshaller)
-            except DataUnmarshallerError:
-                event = from_json(msg, data_unmarshaller=pickle.loads)
-
-            event_logger.debug("handle_dispatch: %s", event)
-
-            await self._registry.all_experiments[0].dispatch(event, 0)
+                # all Protobuf objects come in DispatcherMessage container
+                # which needs to be parsed
+                event = DispatcherMessage()
+                try:
+                    event.ParseFromString(msg)
+                except DecodeError:
+                    logger.error(f"Cannot parse pbuf event: {msg.decode()}")
+                    raise
```

Comment:
`raise from DecodeError` or something else specific?

Reply:
when calling `raise` it will just re-raise the same error and we do log the error, so I guess that's fine :shrug: ?